### PR TITLE
utils/partition.py: Fix fetching the exact mount point

### PR DESCRIPTION
**Pull request by clebergnu**
Patch fixes the partition library by matching both the device and
the mountpoint. This is required because all recent kernels
support device mounting multiple times as well as same mount point
being used for different devices

E.g:
/dev/pmem0 on /mnt/pmem type ext4 (rw,relatime,seclabel)
/dev/pmem0 on /pmem_map type ext4 (rw,relatime,seclabel)
/dev/pmem1 on /pmem_map type ext4 (rw,relatime,seclabel)

Also many generic mountpoints like none, tmpfs would need this for
a cleaner and proper unmount.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by clebergnu** [comment]
@harish-24 what's hapenning here is that some tests use the "loop_size" parameter to the `avocado.utils.partition.Partition` constructor.  With that, the device that is visible under `/proc/mounts` is no longer the "device" (a file, really) given to the constructor, but becomes a `/dev/loopX` device.  Example:

```
/dev/loop0 /tmp/avocado__selftests.unit.test_utils_partition__TestPartition__setUp__zklt_rke/disk ext2 rw,seclabel,relatime 0 0
```

With the change you're proposing, the moutpoint is found, but the device is never found.  Thus, `get_mountpoint()` returns None, which results in this type of test failure:

```
======================================================================
FAIL: test_basic (selftests.unit.test_utils_partition.TestPartition)
Test the basic workflow
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/src/avocado/selftests/unit/test_utils_partition.py", line 64, in test_basic
    with open("/proc/mounts") as proc_mounts_file:
AssertionError: '/tmp/avocado__selftests.unit.test_utils_[42 chars]disk' != None
```

**Review by beraldoleal — APPROVED** [review]

**Comment by clebergnu** [comment]
@beraldoleal FIY there are test breakages here (even with the rebase) and they need to addressed before merging it.

**Comment by beraldoleal** [comment]
> @beraldoleal FIY there are test breakages here (even with the rebase) and they need to addressed before merging it.

Yes, I have noticed that right after the approval. That is the reason I removed the approval.

**Comment by clebergnu** [comment]
Closing in favor of #4439

**Pull request closed without merging**
